tests/api/test_result.py

```diff
@@ -29,7 +27,7 @@ class TestResult:
 
     def test_stateless_result_raises_on_state_access(self):
         """Tests that `result.state` raises an error for a stateless result."""
-        result = Result(np.array([[1, 2], [3, 4]]), is_stateful=False)
+        result = Result({"samples": np.array([[1, 2], [3, 4]])})
```

Comment:
For this test and the ones below:
```suggestion
        result = Result({"output": [np.array([[1, 2], [3, 4]])]})
```